resources/utils/numbers.ts

```diff
@@ -15,13 +15,18 @@ const digitsLookup = [
   { value: 1e18, symbol: 'quintillion' }
 ]
 
-export function formatNumber(n: number, digits = 2) {
+export function formatNumber(n: number, digits = 4) {
   const num = Number(n)
   const item = digitsLookup
     .slice()
     .reverse()
     .find((item) => num >= item.value) || { value: 0, symbol: '?' }
-  const formatted = (value: number) => `${value.toFixed(digits).replace(numberRegex, '$1')} ${item.symbol}`
+
+  const formatted = (value: number) => {
+    const isAproximate = value.toFixed(digits) !== value.toString(10)
```

Comment:
is this logic to handle the unlimited case? it doesn't seem like it's really broadly applicable to many other cases so we might want to just handle the "unlimited" case wherever it happens and keep this function concerned with formatting a number

Reply:
that makes sense. can we allow decimal approvals for tokens?